tools/fuzzer/constraints_test.cc

```diff
@@ -176,3 +176,33 @@ TEST(Constraints, Unsatisfiability) {
   EXPECT_THAT(default_ctor.satisfiable(), IsFalse());
   EXPECT_THAT(default_specific_types.satisfiable(), IsFalse());
 }
+
+TEST(Constraints, MemoryConstraints) {
+  auto expect_fields = [](const MemoryConstraints& constraints,
```

Comment:
Maybe you can just define `operator==` for `MemoryConstraints` (here in the test file) and then just use EXPECT_EQ:

```
bool operator==(const MemoryConstraints& lhs, const MemoryConstraints& rhs){ 
    return lhs.must_be_valid() == rhs.must_be_valid() &&
        lhs.required_freedom_index() == rhs.required_freedom_index();
}
...
EXPECT_EQ(invalid.from_address_of(), {false, 0});
```

Reply:
Nice suggestion, that should improve precision of test reports.
However, the change wasn't as smooth as expected :/ Using `{false, 0}` in `EXPECT_EQ` resulted in compile error because of macro limitations (apparently, wrapping `{}` expression in parens seems to be a common workaround, but didn't work in this case). Even `EXPECT_EQ(lhs, MemoryConstraint(false, 0))` didn't compile.
I decided to use the form `EXPECT_TRUE(lhs == MemoryConstraints(false, 0))`. For constructors I check each field explicitly (to avoid possible modifications by constructor).

Reply:
`EXPECT_EQ(lhs, MemoryConstraints(false, 0))` works after all. Equality comparator just had to be part of the `fuzzer` namespace.